pkg/security/auth.go

```diff
@@ -33,18 +33,21 @@ type UserAuthHook func(context.Context, SQLUsername, bool) (connClose func(), _
 // SetCertPrincipalMap sets the global principal map. Each entry in the mapping
 // list must either be empty or have the format <source>:<dest>. The principal
 // map is used to transform principal names found in the Subject.CommonName or
-// DNS-type SubjectAlternateNames fields of certificates.
+// DNS-type SubjectAlternateNames fields of certificates. This function splits
+// each list entry on the final colon, allowing <source> to contain colons.
 func SetCertPrincipalMap(mappings []string) error {
 	m := make(map[string]string, len(mappings))
 	for _, v := range mappings {
 		if v == "" {
 			continue
 		}
 		parts := strings.Split(v, ":")
```

Comment:
Do you think that `strings.LastIndexByte` would make this shorter and avoid some heap allocations?